runtime/lib/app/ext-app.js

```diff
@@ -6,9 +6,13 @@ var childProcess = require('child_process')
 var path = require('path')
 var logger = require('logger')('ext-app')
 var _ = require('@yoda/util')._
-var ActivityDescriptor = require('./activity-descriptor').ActivityDescriptor
 
-var entry = path.join(__dirname, '..', '..', 'client', 'ext-app-entry.js')
+var kAppModesTest = require('../../constants').AppScheduler.modes.test
+var ActivityDescriptor = require('../descriptor/activity-descriptor')
+var ActivityTestDescriptor = require('../descriptor/activity-test-descriptor')
+
+var defaultEntry = path.join(__dirname, '..', '..', 'client', 'ext-app-entry.js')
+var testEntry = path.join(__dirname, '..', '..', 'client', 'ext-test-entry.js')
```

Comment:
```js
var entryRoot = path.join(__dirname, '..', '..', 'client')
var defaultEntry = entryRoot + '/ext-app-entry.js'
var testEntry = entryRoot + '/ext-test-entry.js'
```

The above would reduce calls to join paths.

Reply:
If I got your pointer right, do you mean:
```
var entryRoot = path.join(__dirname, '..', '..', 'client')
var defaultEntry = path.join(entryRoot, '/ext-app-entry.js')
var testEntry = path.join(entryRoot, '/ext-test-entry.js')
```

Reply:
No need to join for the entries, the function in ShadowNode takes more resources than string concatenation.